.github/workflows/ci_build_android.yml

```
@@ -0,0 +1,53 @@
+name: Build Android library
+
+on:
+  workflow_call:
+    outputs:
+      artifact-dir-base:
+        description: The base name of artifact directory (without the target suffix)
+        value: ${{ jobs.build-android-libs.outputs.artifact-dir-base }}
+
+jobs:
+  build-android-libs:
+    runs-on: ubuntu-20.04
+    timeout-minutes: 35
+    strategy:
+      matrix:
+        target: [
+          { rust: "aarch64-linux-android", ndk: "arm64-v8a" },
+          { rust: "armv7-linux-androideabi", ndk: "armeabi-v7a" },
+          { rust: "x86_64-linux-android", ndk: "x86_64" },
+          { rust: "i686-linux-android", ndk: "x86" }
+        ]
+    env:
+      ARTIFACT_DIR_BASE: ${{ github.job }}
+    outputs:
+      artifact-dir-base: ${{ env.ARTIFACT_DIR_BASE }}
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      - name: Setup the rust CI.
+        uses: ./.github/actions/setup-job
+        with:
+          rust: true
+          android: true
+          rust-target: ${{ matrix.target.rust }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Android library
+        run: |
+          just compile-android "${{ matrix.target.ndk }}"
+
+      - name: Prepare lib for upload
+        id: lib
+        run: |
+          FILE=${{ env.FLUTTER_WORKSPACE }}/android/src/main/jniLibs/${{ matrix.target.ndk }}/libxayn_discovery_engine_bindings.so
```

i thing i forgot. to preserve the folder that cargo ndk creates for us we should change this to 
```suggestion
          FILE=${{ env.FLUTTER_WORKSPACE }}/android/src/main/jniLibs
```
so it will upload 
```
<arch_folder>/lib.so
```
instead of 
```
lib.so
```

we can then also remove this step and put this line directly to the upload action path

---

Done